marklogic-data-hub/src/main/java/com/marklogic/hub/impl/StepDefinitionManagerImpl.java

```diff
@@ -68,6 +73,7 @@ public void saveStepDefinition(StepDefinition stepDefinition, boolean autoIncrem
         } catch (IOException e) {
             throw new DataHubProjectException("Could not write Step to disk for project.");
         }
+        getArtifactService().setArtifact("stepDefinitions", stepDefinition.getName(),mapper.valueToTree(stepDefinition));
```

Comment:
The duplication of calling "valueToTree" here to convert an object into a JsonNode suggests the concept of an abstract Artifact class that Mapping, StepDefinition, EntityModel, and Flow all extend. Artifact would have an e.g. "JsonNode toJsonNode()" method on it. That encapsulates the logic of converting an object to a JsonNode into the object itself, as opposed to exposing and duplicating it across N Manager classes.

Reply:
Is it ok to move the convertArtifactToJson method to JSONUtils so that all the Manager class can use it ?